server/src/main/java/org/opensearch/action/admin/cluster/hahealth/ClusterHaHealthRequest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.hahealth;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.clustermanager.ClusterManagerNodeReadRequest;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Transport request for requesting cluster HA health
+ *
+ * @opensearch.internal
+ */
+public class ClusterHaHealthRequest extends ClusterManagerNodeReadRequest<ClusterHaHealthRequest> {
```

Comment:
Can we rather extend from `ClusterHealthRequest`? That might help to get the capabilities of generic health requests like specifying indices, wait for status, etc. Although those extensions can be taken up in a subsequent PR

Reply:
I thought so but than segregating it seems better because if tomorrow we need to add new capabilities to this API itself would be better and more easy. So thinking in terms of single responsibility model this seems better to me.